pymongo/mongo_client.py

```diff
@@ -2334,6 +2338,8 @@ def run(self) -> T:
                             raise
                         self._retrying = True
                         self._last_error = exc
+                        if self._client.topology_description.topology_type == TOPOLOGY_TYPE.Sharded:
+                            self._deprioritized_servers.append(self._server)
```

Comment:
Since this logic is the same for reads and writes and we do it in one place?

Reply:
Good catch, thanks!